src/main/java/uk/gov/hmcts/ccd/domain/types/DocumentValidator.java

```diff
@@ -74,24 +96,128 @@ public BaseType getType() {
+            validationResults = validateCategoryId(dataFieldId,dataValue,caseFieldDefinition);
+            if (!validationResults.isEmpty()) {
+                return validationResults;
+            }
+        }
 
-            if (!documentBinaryUrlMatcher.matches()) {
-                LOG.error("{} does not match Document Management domain or expected URL path {} {}",
-                          documentUrlValue, urlPatternString, dataFieldId);
-                return Collections.singletonList(new ValidationResult(documentBinaryUrlValue
-                    + " does not match Document Management domain or expected URL path", dataFieldId));
+        if (dataValue.has(UPLOAD_TIMESTAMP)) {
+            final JsonNode uploadTimeStamp = dataValue.get(UPLOAD_TIMESTAMP);
+            validationResults = validateUploadTimeStamp(dataFieldId,uploadTimeStamp,caseFieldDefinition);
+            if (!validationResults.isEmpty()) {
+                return validationResults;
             }
         }
 
         return Collections.emptyList();
     }
 
+    private List<ValidationResult> validateDocumentBinaryURL(final String dataFieldId,
+                                                             final JsonNode documentBinaryUrl,
+                                                             Pattern urlPattern,
+                                                             String documentUrlValue,
+                                                             String urlPatternString) {
+        isNullOrEmptyResult = isNullOrEmpty(documentBinaryUrl,DOCUMENT_BINARY_URL,dataFieldId);
+        if (!isNullOrEmptyResult.isEmpty()) {
+            return isNullOrEmptyResult;
+        }
+
+        final String documentBinaryUrlValue = documentBinaryUrl.textValue();
+        final Matcher documentBinaryUrlMatcher = urlPattern.matcher(documentBinaryUrlValue);
+
+        if (!documentBinaryUrlMatcher.matches()) {
+            LOG.error("{} does not match Document Management domain or expected URL path {} {}",
+                documentUrlValue, urlPatternString, dataFieldId);
+            return Collections.singletonList(new ValidationResult(documentBinaryUrlValue
+                + " does not match Document Management domain or expected URL path", dataFieldId));
+        }
+        return Collections.emptyList();
+    }
+
+    private List<ValidationResult> validateCategoryId(final String dataFieldId,
+                                                      final JsonNode dataValue,
+                                                      final CaseFieldDefinition caseFieldDefinition) {
+        final JsonNode categoryId = dataValue.get(CATEGORY_ID);
+        isNullOrEmptyResult = (isNullOrEmpty(categoryId,CATEGORY_ID,dataFieldId));
+        if (!isNullOrEmptyResult.isEmpty()) {
+            return isNullOrEmptyResult;
+        }
+
+        if (!dataValue.has("parent_category_id")) {
+            LOG.warn("Object does not have parent_category_id key specified");
+        }
```

Comment:
I don't recall the `parent_category_id` being mentioned as ever expected - may want to discuss this one
```suggestion
```

Reply:
Yup, not needed. Got mixed in when looking at another ticket